docs/Lowering.md

```diff
@@ -0,0 +1,230 @@
+The SPIRV-LLVM-Translator will "lower" some LLVM intrinsic calls to another function or implementation
```

Comment:
Lets rename the file from Lowering to LLVMIntrinsicFunctionsLowering or something similar